### PR TITLE
modernize `flutter drive` w.r.t. debug/profile/release options

### PR DESCRIPTION
**Pull request by yjbanov**
- Reject `--release` explicitly as the driver requires observatory
- Pass build mode to APK builder

/cc @devoncarew

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/drive.dart [hunk 1]
```diff
@@ -100,7 +100,17 @@ class DriveCommand extends RunCommandBase {
 
     if (!argResults['use-existing-app']) {
       printStatus('Starting application: ${argResults["target"]}');
-      int result = await appStarter(this, getBuildMode());
+
+      if (getBuildMode() == BuildMode.release) {
+        printError(
+          'Flutter Driver does not support running in release mode. '
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/drive.dart]: do you want a newline at the end of this line and the next line?

Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/drive.dart]: And great, clear messaging here :)

**Comment by devoncarew** [comment]
lgtm

File: packages/flutter_tools/lib/src/commands/drive.dart (hunk 1 above)
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/drive.dart]: DOne

Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/commands/drive.dart]: As-is is fine, but another phrasing: "Flutter Drive requires Observatory which is not available in --release mode, please use --profile".

Comment (eseidelGoogle) [inline packages/flutter_tools/lib/src/commands/drive.dart]: Oh, mb, I see you explain the other two below.

**Pull request merged**